compiler/src/dotty/tools/dotc/typer/Dynamic.scala

```diff
@@ -186,7 +186,7 @@ trait Dynamic {
       // ($qual: Selectable).$selectorName("$name")
       val base =
         untpd.Apply(
-          untpd.TypedSplice(selectable.select(selectorName)).withSpan(fun.span),
+          untpd.Select(untpd.TypedSplice(selectable), selectorName).withSpan(fun.span),
```

Comment:
That will indeed allow extension methods, but won't it also allow all sorts of other things? Like recursive structural lookups and insertion of implicit conversions? That seems too much (especially the recursive structural lookup).

Reply:
It will allow implicit conversions, and that's intentional. It will allow recursive structural lookup _if_ selectDynamic is defined in the structural type. But I think that's a weird corner case, and I also see nothing wrong to support it.

Reply:
EDIT: That should have been `applyDynamic`.